cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/ProvisionResponse.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2021 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.analyzer;
+
+import static com.linkedin.cruisecontrol.common.utils.Utils.validateNotNull;
+
+
+/**
+ * Indicates the {@link ProvisionStatus} along with the recommended actions regarding the relevant status.
+ * Recommendations are only relevant to {@link ProvisionStatus#UNDER_PROVISIONED} and {@link ProvisionStatus#OVER_PROVISIONED}.
+ */
+public class ProvisionResponse {
+  private ProvisionStatus _status;
+  private final StringBuilder _recommendation;
```

Comment:
Can the recommendations ever be duplicated?  In which case maybe this should be Set<String>?  If this a human readable string then the different recommendation may need to be formatted.  For example adding new lines in the right places; that implies some collections of String.

Reply:
Recommendations are generated by goals. In theory, they can be duplicated (i.e. different goals might recommend the same action), but in practice recommendations include identifiers (i.e. a goal name prefix), which makes them unique across goals.

I updated the constructor with a new `recommender` field to make it more clear that the recommender is required in addition to recommendation.

> If this a human readable string then the different recommendation may need to be formatted. For example adding new lines in the right places; that implies some collections of String.

`ProvisionResponse #aggregateRecommendations(ProvisionResponse)` does this formatting using a space as the delimiter between recommendations -- I don't use "new line" as the delimiter, because I feel that it makes the overall response harder to read / parse -- e.g. when used in json response field `provisionRecommendation` of `OptimizerResult`.